CompiletimeCrashTests/check.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+from sys import argv, stdin, stderr
+from enum import Enum
+from typing import Self
+import re
+
+class ReproducerType(Enum):
+    OK = 0      # should build successfully
+    XERROR = 1  # expected to throw compilation error with proper diagnostics
+    CRASH = 2   # crashes
+    ERROR = 3   # should build successfully, but throws compilation error
+
+    @staticmethod
+    def parse(header: str) -> type[Self | None]:
+        match = re.match(r"^#\s*(OK|ERROR|CRASH|XERROR)\b", header)
+        if match:
+            return ReproducerType[match.group(1)]
+        return None
+
+
+def errprint(*args, **kwargs):
+    print(*args, **kwargs, file=stderr)
+
+def check(expected: [str], found: [str]):
+    reproducer_type = ReproducerType.parse(expected[0])
+    if reproducer_type:
+        expected = expected[1:]
+    if not expected:
+        if not found:
+            exit(0)
+        errprint(f"FAIL: expected no output, found {found}")
+        exit(1)
+    idx = 0
+    expected_line = expected[idx]
+    for line in found:
+        if expected_line in line:
+            idx += 1
+            if idx == len(expected):
+                exit(0)
+            expected_line = expected[idx]
+        if idx == len(expected):
+            exit(0)
+    errprint(f"FAIL: \"{expected_line} not found in {found}\"")
```

Comment:
This doesn't cover the case when there's no header but there are expected lines in a file. Would be nice to catch in case someone forgets to add the header :)!